msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/JwtHelper.java

```diff
@@ -56,11 +56,12 @@ static ClientAssertion buildJwt(String clientId, final ClientCertificate credent
                 builder.x509CertChain(certs);
             }
 
-            //SHA-256 is preferred, however certain flows still require SHA-1 due to what is supported server-side
-            if (useSha1) {
-                builder.x509CertThumbprint(new Base64URL(credential.publicCertificateHashSha1()));
+            //SHA-256 is preferred, however certain flows still require SHA-1 due to what is supported server-side. If SHA-256
+            // is not supported or the IClientCredential.publicCertificateHash256() method is not implemented, the library will default to SHA-1.
+            if (useSha1 || credential.publicCertificateHash256() == null) {
```

Comment:
You're using `credential.publicCertificateHash256()` twice - once in the if check and once in the else. Can you set it to a const before the if check?

(and add a type, below is typescript)
```typescript
const publicCertificateHash256: hash | null = credential.publicCertificateHash256();
```

Reply:
In the latest commit it's now only set once.